File: rBike.Model/Comment.cs
using System;

namespace rBike.Model
{
    public class Comment
    {
        public int CommentId { get; set; }

        public string? Content { get; set; }

        public DateTime? DateAdded { get; set; }

        public int UserId { get; set; }

        public int BikeId { get; set; }

        public string? Status { get; set; }

        public virtual Bike Bike { get; set; } = null!;

        public virtual User User { get; set; } = null!;

        public string? BikeName { get; set; }

        public string? Username { get; set; }
    }
}

File: UI/rbike_admin/lib/models/comment.dart
import 'package:json_annotation/json_annotation.dart';

part 'comment.g.dart';

@JsonSerializable()
class Comment {
  int? commentId;
  String? content;
  DateTime? dateAdded;
  int? userId;
  int? bikeId;
  String? status;
  String? bikeName;
  String? username;

  Comment();

  factory Comment.fromJson(Map<String, dynamic> json) => _$CommentFromJson(json);

  Map<String, dynamic> toJson() => _$CommentToJson(this);
} 